src/main/java/seedu/address/model/event/Event.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.event;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.model.person.Person;
+
+/**
+ * Represents an Event in the address book.
+ */
+public class Event {
+    private String title;
+    private String description;
+    private LocalDateTime dateTime;
+    private ArrayList<Duration> reminderDeltas;
+    private Set<Person> members;
+
+    /**
+     * Initialises {@code Event} instance.
+     * @param title Title of event.
+     * @param description Description of event.
+     * @param dateTime Date and time of event.
+     */
+    public Event(String title, String description, LocalDateTime dateTime) {
+        this.title = title;
+        this.description = description;
+        this.dateTime = dateTime;
+        this.reminderDeltas = new ArrayList<>();
+        this.members = new HashSet<>();
+    }
+
+    public void addMember(Person member) {
+        members.add(member);
+    }
+
+    public void addReminder(Duration reminderDelta) {
+        reminderDeltas.add(reminderDelta);
+    }
+
+    private boolean shouldTriggerReminder(LocalDateTime currentDateTime) {
```

Comment:
Perhaps use a better choice of boolean method name to follow the recommended Java Coding standard by starting with `is`, `has`, `was`? Perhaps `isTriggered`?